core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -278,17 +273,6 @@ private synchronized void initContext(ClientContext ctx,
     mBlockMasterClientPool = new BlockMasterClientPool(mMasterClientContext);
     mBlockWorkerClientPoolMap = new ConcurrentHashMap<>();
     mUriValidationEnabled = ctx.getUriValidationEnabled();
-
-    MetricsSystem.registerGaugeIfAbsent(TOTAL_RPC_CLIENTS_METRICS_NAME,
-       new CachedGauge<Integer>(1, TimeUnit.MINUTES) {
-          @Override
-          protected Integer loadValue() {
-            int totalClients = mFileSystemMasterClientPool.size() + mBlockMasterClientPool.size();
-            totalClients += mBlockWorkerClientPoolMap.values()
-                .stream().mapToInt(DynamicResourcePool::size).sum();
-            return totalClients;
-          }
-        });
```

Comment:
Although I didn't directly find the leak in the customer environment from here, I suspect this one also leaks references to the pool. @dbw9580 do you know if using a lambda will leak refs but using anonymous inner class like this is fine? 

Regardless of the leak concerns, I'm not sure the metric is correct here. Is it true that one Alluxio Client will have only one FileSystemContext? If a client has multiple FileSystemContext instances, would the gauge be incorrect?

Reply:
> using anonymous inner class like this is fine?

No. Inner classes capture a reference to the enclosing class even the reference is not used. 

ref: https://stackoverflow.com/a/8590414